src/test/kotlin/io/zerobase/smarttracing/ApiIT.kt

```diff
@@ -162,6 +176,59 @@ class ApiIT {
         assertThat(scannables).isNotNull.isNotEmpty.hasSize(1).first().extracting("id", "name").contains(scannableId, "New Name")
     }
 
+    @Test
+    fun shouldConnectBothEdgesOnSelfReportedTest() {
+        val deviceId = createFake("Device")
+
+        val reportId: String? = UriBuilder.fromUri("http://localhost:${app.getPort(0)}")
+            .path("/devices/{id}/reports/tests")
+            .build(deviceId)
+            .let { app.client().target(it) }
+            .request()
+            .post(
+                Entity.json(SelfReportedTestResult(LocalDate.now().minusDays(1), false, Instant.now())),
+                idWrapperType
+            )["id"]
+
+        assertThat(reportId).isNotNull()
+
+        val report: Map<Any, Any> = g.V(reportId).valueMap<Any>().with(WithOptions.tokens).by(unfold<Any>()).next()
+
+        assertThat(report)
+            .containsEntry(T.label, "TestResult")
+            .containsEntry("verified", false)
+            .containsEntry("testDate", LocalDate.now().minusDays(1).toString())
+
+        val otherVertexes = g.V(reportId).hasLabel("TestResult").bothE("REPORTED", "REPORT_FOR").otherV().id().toList()
+        assertThat(otherVertexes).isNotNull.isNotEmpty.hasSize(2).containsOnly(deviceId)
+    }
+
+    @Test
+    fun shouldConnectBothEdgesOnSelfReportedSymptoms() {
```

Comment:
would it be worth to also check within entries like the entry "Symptoms" also contains the properties (like .symptoms that I had noticed was missing before, though unsure if that was on purpose) or is that too specific?

Reply:
yes, it's very much worth checking those. as i mentioned before, i completely spaced the symptoms field.